opentreemap/treemap/udf.py

```diff
@@ -630,10 +674,7 @@ def _validate_single_datatype(self, datatype):
             raise ValidationError(_('invalid datatype'))
 
         if datatype['type'] in ('choice', 'multichoice'):
-            choices = datatype.get('choices', None)
-
-            if choices is None:
-                raise ValidationError(_('Missing choices key'))
```

Comment:
This line in `_validate_and_update_choice` will raise a `KeyError` if the datatype does not have a `choices` key.
https://github.com/OpenTreeMap/otm-core/pull/3036/files#diff-5232da98c01c519f4805c4259ad42fe1R390

It would be safer if we kept choices as a required field and used and empty list in the case where all choices are protected.